zqd/space/archivespace.go

```diff
@@ -0,0 +1,162 @@
+package space
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zqd/api"
+	"github.com/brimsec/zq/zqd/storage"
+	"github.com/brimsec/zq/zqd/storage/archivestore"
+	"github.com/brimsec/zq/zqe"
+)
+
+type archiveSpace struct {
+	spaceBase
+
+	// confMutex protects changes to configuration changes.
+	confMutex sync.Mutex
```

Comment:
Nit: Our conventional name for a sync.Mutex or sync.RWMutex field is mu or whateverMu, so call this confMu. You might also consider grouping it separately with the conf field to emphasize that it protects conf but not path.